encord/client.py

```diff
@@ -504,10 +533,74 @@ def add_private_data_to_dataset(
         else:
             raise ValueError(f"Type [{type(private_files)}] of argument private_files is not supported")
 
-        payload = {"files": files, "integration_id": integration_id, "ignore_errors": ignore_errors}
-        response = self._querier.basic_setter(DatasetData, self._config.resource_id, payload=payload)
+        process_hash = self._querier.basic_setter(
+            DatasetDataLongPolling,
+            self._config.resource_id,
+            payload={
+                "files": files,
+                "integration_id": integration_id,
+                "ignore_errors": ignore_errors,
+            },
+        )["process_hash"]
+
+        logger.info(f"add_private_data_to_dataset job started with upload_job_id={process_hash}.")
+        logger.info("SDK process can be terminated, this will not affect successful job execution.")
+        logger.info("You can follow the progress in the web app via notifications.")
+
+        return process_hash
+
+    def add_private_data_to_dataset_get_result(
```

Comment:
I think the original design by @alexey-cord-tech had something like a future in mind, like 
```py
awaitable_object =  dataset.add_private_data_to_dataset_result()
result = awaitable_object.await()
```

Reply:
probably we want to call this `get_async_private_cloud_upload_result`. No need to keep the `private_data_to_dataset` name around which is not great imo.

Reply:
I would vote against awaitable object. In my opinion having a upload_job_id return and log in console will be simpler for a person which want's to start a job, copy past upload_job_id somewhere, come back after 2h and check progress by just calling correct function with upload_job_id string.

What would be advantage of using awaitable object? (Fact that we store upload_job_id as object attribute instead of storing a string?)

Imo using object here could be misleading if we additionaly want to store job state in this object (or maybe you guys are not proposing this?), because state locally will need to be synced with state on BE.

Functional approach instead is imo simpler, you have upload_job_id, you want state? Call this function and get it? Oh you have it? Nice

Reply:
@sergei-encord let's not use async keyword because it means something particular in python world.
I think we should keep the old prefix `add_private_data_to_dataset` with some postfix, @denis-cord wdyt?

At the end of the day yes we could change `add_private_data_to_dataset` prefix to something different, but we are basically doing add_private_data_to_dataset in 2 parts, so having direct naming connection to the old thing which was doing 2 at the same time would be imo nice?

Reply:
Agree with the point that we shouldn't not overuse async word. So +1 for keeping current naming.

Reply:
@alexey-cord-tech I have included full interface in Notion (calling functions with string ids)

Reply:
I have to say that having 
1) `add_private_data_to_dataset`
2) `add_private_data_to_dataset_start`
3) `add_private_data_to_dataset_get_result`

Is confusing to me. If 2 has "start", then what is 1) ?

I'd say that adding `async` to the name doesn't abuse the "keyword" because we're not using keywords here. But I'm happy with the consensus, just find it personally weird. 

It will be important also in 1) to update the docstrings and prompt the user to use 2 and 3 instead.